Utils/src/FastSimWeightPR31285To36122.cc

```diff
@@ -0,0 +1,246 @@
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq4->SetBinContent(20,2.2);
+   
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5 = new TH1F("hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5","",20,0,500);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(1,1.010584);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(2,1.01473);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(3,1.071903);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(4,1.192592);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(5,1.332403);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(6,1.573248);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(7,1.732668);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(8,2.013557);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(9,2.270501);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(10,2.578063);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(11,2.806123);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(12,3.175);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(13,3.385366);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(14,3.609589);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(15,4.896341);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(16,4.672131);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(17,5.119048);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(18,5.574074);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(19,6.516129);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(20,7.685393);
+   
+   
+}
+
+FastSimWeightPR31285To36122::~FastSimWeightPR31285To36122() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+
+// ------------ method called for each event  ------------
+void FastSimWeightPR31285To36122::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+    using namespace edm;
+
+    double EventWeight = 1.0;    
+    
+    edm::Handle< View<reco::GenParticle> > genPartCands;
+    iEvent.getByToken(genCollectionTok, genPartCands);
+    
+    edm::Handle< View<reco::GenJet> > genJets;
+    iEvent.getByToken(genJetTok, genJets);
+    
+    edm::Handle< View<pat::Jet> > recJets;
+    iEvent.getByToken(recJetTok, recJets);   
+    
+    
+    bool jetHasOffendingGp_;
+    TLorentzVector genJet_tlv;
+    TLorentzVector recJet_tlv;
+    TLorentzVector genPart_tlv;    
+    float dr;    
+    float mindr;
+    double leadHadronPt;
+    float originxy;   
+    float decayxy;       
+    int hadronFlavor;     
+    int pdgid;
+    
+
+    for(const auto& genJet : *genJets) 
+    {
+           if (!(genJet.pt()>30)) continue;
+           genJet_tlv.SetPxPyPzE(genJet.px(), genJet.py(), genJet.pz(), genJet.energy());
```

Comment:
TLorentzVectors are unnecessary here. All CMSSW candidate object have `.p4()` accessors that return a more efficient 4-vector.